### PR TITLE
Fixes for running the demo with push mode

### PR DESCRIPTION
**Pull request by davidfestal**
This PRs provides additional fixes to enable using the push mode to run the demo.
It's WIP and might need to be more tested / completed before merging.

### CODE REVIEW

File: contrib/demo/script [hunk 1]
```diff
@@ -43,110 +44,65 @@ clear
 pe "kubectl config view --minify | grep server:"
 pe "kubectl config use-context user"
 pe "kubectl config view --minify | grep server:"
-p "kubectl get databases"
-echo "error: the server doesn't have a resource type \"databases\""
+pe "kubectl get databases"
+#echo "error: the server doesn't have a resource type \"databases\""
```
Comment (imjasonh) [inline contrib/demo/script:48]: There's some leftover commented out stuff here and below.

File: pkg/syncer/syncer.go [hunk 2]
```diff
@@ -222,9 +222,18 @@ func (c *Controller) process(gvr schema.GroupVersionResource, obj interface{}) e
 	klog.V(2).Infof("Process object of type: %T : %v", obj, obj)
 	meta, isMeta := obj.(metav1.Object)
 	if !isMeta {
-		err := fmt.Errorf("Object to synchronize is expected to be a metav1.Object, but is %T", obj)
-		klog.Error(err)
-		return err
+		if tombstone, isTombstone := obj.(cache.DeletedFinalStateUnknown); isTombstone {
+			meta, isMeta = tombstone.Obj.(metav1.Object)
+			if !isMeta {
+				err := fmt.Errorf("Tombstone contained object that is not expected %#v", obj)
```
Comment (imjasonh) [inline pkg/syncer/syncer.go:228]: Can you log the expected and actual types here too?

**Review by imjasonh — APPROVED** [review]

File: contrib/demo/script (hunk 1 above)
Comment (davidfestal) [reply contrib/demo/script:48]: I kept the expected output here on purpose, at least for now.

Comment (davidfestal) [reply contrib/demo/script:48]: But let me remove them on every call that now fully produces the expected result.

**Pull request merged**